app/views/snps/show.html.erb

```diff
@@ -59,6 +63,22 @@
       We don't have enough users yet.
     <%end%>
   </div>
+
+  <div class="span3 columns">
+    <center><h6><a href="#" rel="tooltip" data-html="true" title="Phenotypes possibly associated with this SNP.">Phenotypes</a></h6></center>
+    <% if @snp.phenotypes.count %>
+      <table id="phenotype_table" class="table table-striped">
+        <% @ordered_phenotype_snps.each do |p| %>
+          <tr>
+            <td> <%= Phenotype.find(p.phenotype_id).characteristic %> </td>
```

Comment:
This sends a query to the database for every iteration of the loop. You can get the phenotypes all at once, by using `includes` on `PhenotypeSnp`:

```rb
@ordered_phenotype_snps = @snp.phenotype_snps.includes(:phenotype).order(score: :desc)
@ordered_phenotype_snps.each do |phenotype_snp|
  phenotype_snp.phenotype.characteristic
end
```